pytest_pyvista/pytest_pyvista.py

```diff
@@ -28,23 +30,90 @@
 VISITED_CACHED_IMAGE_NAMES: set[str] = set()
 SKIPPED_CACHED_IMAGE_NAMES: set[str] = set()
 
-
-def _get_env_info() -> str:
-    system = platform.system()
-    if system == "Darwin":
-        system = "macOS"
-
-    return "_".join(
-        [
-            f"{system}-{platform.release()}",
-            f"py-{sys.version_info.major}.{sys.version_info.minor}.{sys.version_info.micro}",
-            f"pyvista-{pyvista.__version__}",
-            f"vtk-{vtkmodules.__version__}",
+_GPU_VENDOR: list[str] = [""]  # Use a list so we can mutate the string globally
+
+
+@dataclass
+class _EnvInfo:
+    prefix: str = ""
+    os: bool = True
+    machine: bool = True
+    python: bool = True
+    pyvista: bool = True
+    vtk: bool = True
+    gpu: bool = True
+    ci: bool = True
+    suffix: str = ""
```

Comment:
If a user wants to override everything, he/she needs to set `False` to all attributes of the instance.
Maybe we could add a `custom` attribute to simplify this.

```suggestion
    suffix: str = ""
    custom: str = ""
```

Reply:
We could also allow `EnvInfo | str` for this use case. And just set it directly to the desired string and bypass the dataclass entirely.

Reply:
This was already possible, (`str(self.env_info)` is used), but tests/docs are added in [79d6f59](https://github.com/pyvista/pytest-pyvista/pull/214/commits/79d6f59a8a3ece39dc7adc6c0a828d7673d9d200)